### ISSUE TITLE
Make a release that can be install through pip with String support

### ISSUE THREAD

**Issue by Phat3** [pysmt/pysmt#479]
Hi, 

We implemented String constraint collection in binaries in a symbolic execution framework called [angr](https://github.com/angr), specifically in the component called claripy. The system has already been implemented and tested using `pysmt 0.7.1.dev1` and the code can be find in [this repo](https://github.com/Lukas-Dresel/claripy/tree/feat/strings) on the branch `feat/strings`.

The problem is the following: before merging it into master we need a version of pysmt with string support that can be installed using pip directly without using the link to the github repo ( i. e. using `pip install pysmt` and not using `pip install git+https://github.com/pysmt/pysmt.git`).

Is it possible to rollout this release? If it's not ready yet we are willing to help in the developing process.

Bye.

**Comment by marcogario** [comment]
Thanks for contributing!

Would the release require to include the PR for regex that you submitted? If not, we can make a release in the next few days. Otherwise we need a bit more time to review the PR.

The only open point might be the integration with z3. I know we are using an old version, and we bumped into some issue when upgrading it, but nothing prevents us from keeping the old version. 


@mikand anything that you think needs to be addressed prior to the release?

**Comment by Phat3** [comment]
No, the regex support is not needed for that project... We would probably need it later in the future when we roll out our Java symbolic execution framework but for now is not a priority. 

From the pysmt side we only need to be able to parse and dump constraints related to string theory (the solver part is not needed).

PS: For the regex pull request let me know about problems, bugs, and improvements... I'll be happy to contribute :smile:

**Comment by mikand** [comment]
@marcogario I think we can go ahead ad do a release, unfortunately in the short term I'll have no time to work on this. Maybe a very cheap test would be to to update Z3, but we can safely keep the current version as you suggested.

**Comment by Phat3** [comment]
Thanks! :smile:
